juriscraper/pacer/http.py

```
@@ -0,0 +1,170 @@
+        # training account
+        return 'https://dcecf.psc.uscourts.gov/cgi-bin/login.pl'
+    return 'https://pacer.login.uscourts.gov/csologin/login.jsf?pscCourtId=%s' % court_id
+
+
+def login(court_id, username, password):
+    """
+    Log into a PACER jurisdiction via the main PACER portal which should set our global PacerSession
+
+    :param court_id: id of the court to authenticate with
+    :param username: PACER username
+    :param password: PACER password
+    :return: new PacerSession configured with PacerSession token in cookie
+    """
+    if username == 'tr1234':
+        return _login_training(court_id, username, password)
+
+    url = _make_login_url(court_id)
+    logger.info("Logging into: %s at %s" % (court_id, url))
+
+    login_session = requests.Session()
+    login_session.headers['User-Agent'] = 'Juriscraper'
+    login_session.verify=False
+
+    # initial GET to login page to get JSESSIONID
+    r = login_session.get(url, timeout=60)
+    if not r.status_code == 200:
+        msg = 'Could not navigate to PACER central login url: %s' % url
+        logger.error(msg)
+        raise BadLoginException(msg)
+
+    # with our JSESSIONID, try the login
+    login_data = {
+        'login': 'login',
+        'login:loginName': username,
+        'login:password': password,
+        'login:clientCode': '',
+        'login:fbtnLogin': '',
+        'javax.faces.ViewState': 'stateless'
+    }
+    r = login_session.post(url, timeout=60, data=login_data, allow_redirects=False)
+
+    if r.status_code == 302:
+        # we should be redirected on success with cookies!
+        if not login_session.cookies.get('PacerSession', None, '.uscourts.gov', '/'):
+            logger.error('Failed to get a PacerSession token!')
+            raise BadLoginException('Failed to get a PacerSession token!')
+    else:
+        msg = 'Unknown PACER login error: http status %s' % r.status_code
+        if 'Invalid ID or password' in r.text:
```

I wonder if we should raise an exception here. Upstream we'll want to know what kind of failure we got.